### ISSUE TITLE
annotations getting out of synch with type checker?

### ISSUE THREAD

**Issue by leouk** [overturetool/vdm-vscode#157]
Nick, 

I am noticing a subtle situation that "locks" the type checker in the wrong state. It does seem like something is out of synch. The exact same model type checks on the command line and reports errors on VSCode, even after a refresh or a small change + save. I have to restart VSCode for it to go away. 

The situation seems to be with @OnFail annotation parameters: it said the names were not in scope (despite they being in scope). That is, this is a type error within the @OnFail parameters?

**Comment by nickbattle** [comment]
Hmm. There's definitely something wrong here, because I can't get those @OnFails to report name scope errors at all. It may be linked to the use of RESULT, which is a slightly special case. Or perhaps because it's in a postcondition. But I'll have to debug to find out more. It should definitely reports type errors for @OnFail parameters.

I'll attach your example here:
[test.vdmsl.txt](https://github.com/overturetool/vdm-vscode/files/8692578/test.vdmsl.txt)

**Comment by leouk** [comment]
In my concrete example I wasn't using RESULT, but it was on a post. The example post had some OnFail commented out as things evolved, then suddenly, it said that the parameters were out of scope (when they were in scope).

**Comment by nickbattle** [comment]
This is very strange. I can't get the @OnFails to be recognised at all. But when I run the LSP server in a debugger it all works fine. So it must be something about the classpath, properties or environment in the normal case. I'll try to get diagnostics...

**Comment by nickbattle** [comment]
This might be something to do with the vdmj.properties file. Do you have one of those? Can you try deleting it, or trimming it down to the essentials that you need? I'm seeing something strange where the annotations are not parsing when the file is present (though that doesn't explain what you were seeing exactly).

**Comment by nickbattle** [comment]
Ah! There's a bug in the standard vdmj.properties. It has spaces on the end of the annotations path, which causes the annotation parser to fail. This line:
```
# The package list for annotation classes to load.
# (default "com.fujitsu.vdmj.ast.annotations;annotations.ast")
vdmj.annotations.packages = com.fujitsu.vdmj.ast.annotations;annotations.ast	<--- SPACES ON THE END!
```
After I clear this up, it all seems to behave correctly though. Can you try this and re-test? I'll raise a new bug for it.

**Comment by leouk** [comment]
Ah, okay. I had a vdm.properties with the multi line comment stuff; which didn't work by the way on .vscode/

Will try shortly. 

> Ah! There's a bug in the standard vdmj.properties. It has spaces on the end of the annotations path, which causes the annotation parser to fail. This line:
> 
> ```
> 
> # The package list for annotation classes to load.
> 
> # (default "com.fujitsu.vdmj.ast.annotations;annotations.ast")
> 
> vdmj.annotations.packages = com.fujitsu.vdmj.ast.annotations;annotations.ast	<--- SPACES ON THE END!
> 
> ```
> 
> After I clear this up, it all seems to behave correctly though. Can you try this and re-test? I'll raise a new bug for it.

**Comment by nickbattle** [comment]
It's "vdmj.properties" not "vdm.properties"? The multi-line comment stuff
is this one:

# Enable merging of adjacent line comments into a single block comment.
vdmj.parser.merge_comments = false

Cheers,
-nick

**Comment by leouk** [comment]
Yes, vdmj.properties (typo)!. I was also passing -Dvdmj.parser.merge_comments=false on the script for the command line and ./vscode/vdmj.properties on VSCode folder

**Comment by nickbattle** [comment]
So... just to check... you obviously need to pass "...=true" to enable this, right? False is the default (disabled).

**Comment by leouk** [comment]
Yes. =true

**Comment by nickbattle** [comment]
Let's pick the sync issue up with #160. There was a problem to do with annotation paths with vdmj.properties, but that is now covered by #159. I think we can close this one?
